### PR TITLE
Poisson regression R analyses on crime count data

### PR DESCRIPTION
**Pull request by annahadji**
Poisson regression (and negative binomial regression) analysis for crime data in wrattler notebook.

Other changes include: added text surrounding introduction/explanation of models, minor css style changes to javascript visualisation and factorisation of js functions out into separate helper file which is then loaded.

### CODE REVIEW

File: resources/scenic-analysis.wrattler [hunk 1]
```diff
@@ -1,42 +1,60 @@
...
+<br /> 
+The Broken Window hypothesis proposes that visible signs of crime and civil disorder creates an environment that encourages further crime and disorder.
+The hypothesis has been reponsible for promoting controversial policing policies throughout the 1990s. 
+Perhaps the most notable application of the theory was the adoption of the *stop-and-frisk* practise in New York City. 
+Whilst application of such policies have been linked to decline in crime rates, great skeptism surrounds the validity of such a relationship
+because of other changes taking place at that time, including the number of police officers, periods of economic growth and a reduction in poverty.
 
-## Crime data
+We wish to study the relationship between crime rates and scenicness in areas across London, as well as accounting for other indices of deprivation.
+In this notebook we,
```
Comment (illushka) [inline resources/scenic-analysis.wrattler:11]: This is a great right up. Perhaps add, after "We wish to study the relationship between crime rates and scenicness in areas across London, as well as accounting for other indices of deprivation."  "Specifically, we want to test the hypothesis that less scenic areas will be linked to higher rates of crimes." Then we can come back to this later.

File: resources/scenic-analysis.wrattler [hunk 2]
```diff
@@ -97,28 +106,13 @@ if population_density.isnull().values.any():
...
-
-
-
-
-
-
-
-
+Lastly, to obtain a measure of **scenicness** for each LSOA, we load predictions made by a Neural Network that was trained on [Scenic-Or-Not](http://scenicornot.datasciencelab.co.uk/) data on Google Street View images. 
+The Scenic-Or-Not dataset contains around 217,000 images of landscapes that cover nearly 95% of the UK - and has 1.5 million ratings of scenicness from members of the public and counting.
```
Comment (illushka) [inline resources/scenic-analysis.wrattler:116]: Can you add this text "In order to improve the performance for Google Street View images, this CNN is further trained on the 6,946 Google Street View images. " then continue with

"The full method has been discussed in..."

File: resources/scenic-analysis.wrattler [hunk 3]
```diff
@@ -570,15 +496,219 @@ makeFullScreen({title:"Crime/Scenicness Across London LSOAs", height:700}, funct
...
+
+pop_plot <- ggplot(data = subset(long_data, variable=="population_density_area_hectares"), aes(x=mean_monthly_crime_count, y=value)) +
+                   geom_point(size=0.8) + xlim(0, 50) + ylab("population density") + xlab("mean monthly crime count")
+
+living_e_plot <- ggplot(data = subset(long_data, variable=="living_environment_score"), aes(x=mean_monthly_crime_count, y=value)) +
+                        geom_point(size=0.8) + xlim(0, 50) + ylab("living environment score") + xlab("mean monthly crime count")
+
+housing_plot <- ggplot(data = subset(long_data, variable=="barriers_to_housing_and_services_score"), aes(x=mean_monthly_crime_count, y=value)) +
+                       geom_point(size=0.8) + xlim(0, 50) + ylab("barriers to housing and services score") + xlab("mean monthly crime count")
+```
```
Comment (illushka) [inline resources/scenic-analysis.wrattler:568]: This plots are lot more easier to understand. How about also adding a geom_smooth line, some alpha to the points, and a bit of colour so they look nice. Something like:
scenic_plot <- ggplot(data = subset(long_data, variable=="scenic_rating"), aes(x=mean_monthly_crime_count, y=value)) +  geom_point(alpha = 4/20,  color = 'darkorange') + xlim(0, 50) + ylab("scenic rating") + xlab("mean monthly crime count") + geom_smooth(method = 'lm', color="Dark red")

But feel free to choose any colours you like! https://www.nceas.ucsb.edu/~frazier/RSpatialGuides/colorPaletteCheatsheet.pdf

Comment (illushka) [reply resources/scenic-analysis.wrattler:568]: Then, can you add some comments about the relationships we see e.g. mean monthly crime count is higher when scenicness is slightly lower

File: resources/scenic-analysis.wrattler [hunk 4]
```diff
@@ -570,15 +496,219 @@ makeFullScreen({title:"Crime/Scenicness Across London LSOAs", height:700}, funct
...
+By controlling for important covariates, we can obtain more precise estimates of the relationship between scenic rating and number of crimes in an LSOA.
+Above we run three Poisson regression models:
+* `scenic_only_model` - model containing scenic rating only, with no other predictor variables
+* `full model` - model containing scenic rating and population density and indices of deprivation variables
+* `pop_dep_model` - model containing population density and indices of deprivation, excluding scenic rating as a predictor variable
+
+Comparing these models should give an indication of how much information there is by including scenic ratings.
+However, the overdispersion tests suggests that the data has variation that is much higher than would be expected (the rule of thumb is that the ratio of deviance to df should be ~1).
+So below we run a negative binomial regression model which relaxes this *mean == variance* assumption of the Poisson distribution.
+
```
Comment (illushka) [inline resources/scenic-analysis.wrattler:666]: This is really great and so much easier to follow now.

File: resources/scenic-analysis.wrattler [hunk 5]
```diff
@@ -1,42 +1,60 @@
 # Testing the Broken Window hypothesis: are less scenic neighbourhoods linked to more crime?
+<br /> 
+The Broken Window hypothesis proposes that visible signs of crime and civil disorder creates an environment that encourages further crime and disorder.
+The hypothesis has been reponsible for promoting controversial policing policies throughout the 1990s. 
+Perhaps the most notable application of the theory was the adoption of the *stop-and-frisk* practise in New York City. 
+Whilst application of such policies have been linked to decline in crime rates, great skeptism surrounds the validity of such a relationship
+because of other changes taking place at that time, including the number of police officers, periods of economic growth and a reduction in poverty.
 
-## Crime data
+We wish to study the relationship between crime rates and scenicness in areas across London, as well as accounting for other indices of deprivation.
```
Comment (illushka) [inline resources/scenic-analysis.wrattler]: Change to "We wish to study the relationship between crime rates and scenicness in areas across London, while controlling for other factors that could be linked with crimes, such as income and employment deprivation and population density."

File: resources/scenic-analysis.wrattler [hunk 6]
```diff
@@ -97,28 +106,13 @@ if population_density.isnull().values.any():
...
-
-
-
-
-
-
-
-
-
+Lastly, to obtain a measure of **scenicness** for each LSOA, we load predictions made by a Neural Network that was trained on [Scenic-Or-Not](http://scenicornot.datasciencelab.co.uk/) data on Google Street View images.
```
Comment (illushka) [inline resources/scenic-analysis.wrattler]: Add Convolution to Neural Network and (CNN)

**Review by illushka — APPROVED** [review]
Hi Anna.

I think this is looking really great and I have only suggested some minor changes which I am happy for you to do and then merge. The only thing I am wondering is that if the story is getting lost with all the explanations about the different methods and perhaps we might need to bring that back in or move some of the methods to a different section.

For example, in the analyses area, it would be good to re-emphasise why we want to control for other factors that are also linked to crime such as area deprivation variables. Also, it seems intuitive that places with high population density with have higher rates of crime. This gets slightly lost for a while until we get to the conclusion so might be worth another look. However, may be it makes sense to revisit this once we have gathered some feedback from a few people?

In general, though I think this is a great showcase and I hope that demo went really well!
Best wishes,
Chanuki

**Comment by annahadji** [comment]
Hi Chanuki,

Thanks for the feedback/comments! And the demo went really well last week 😃and thank you again for letting us use the data for it.

I've made the changes - and I completely agree that some of the story was getting lost a little along the way so I've added a couple of things that hopefully keeps it tied in/clarifies things.

**Pull request merged**